src/codewhisperer/service/keyStrokeHandler.ts

```diff
@@ -118,6 +118,13 @@ export class KeyStrokeHandler {
         if (!isAutomatedTriggerEnabled) {
             return ''
         }
+
+        /**
+         * Skip when document is reformatted by other extension
+         */
+        if (event.contentChanges.length != 1) {
```

Comment:
1. A code format may only have one contentChange (length = 1)
2. Use !==
3. User input may have more than one contentChange because auto closing bracket. Try input (

Reply:
yep make sense

Reply:
let me figure out and try out other different logics